lib/Server.js

```diff
@@ -973,7 +973,7 @@ class Server {
     // https://github.com/webpack/watchpack/blob/master/lib/DirectoryWatcher.js#L49
     // this isn't an elegant solution, but we'll improve it in the future
     // eslint-disable-next-line no-undefined
-    const usePolling = watchOptions.poll ? true : undefined;
+    const usePolling = !!watchOptions.poll || watchOptions.usePolling;
```

Comment:
Prefer to use `Boolean` instead `!!`

Reply:
yep correct 

```
const usePolling = watchOptions.usePolling || Boolean(watchOptions.poll);
const interval =
  typeof watchOptions.poll === 'number'
    ? watchOptions.poll
    : // eslint-disable-next-line no-undefined
      undefined;
```

this will handle the above cases 👍

Reply:
```
const usePolling = watchOptions.usePolling || Boolean(watchOptions.poll);
const interval =
  typeof watchOptions.internal !== 'undefined'
    ? watchOptions.internal
    : 
      typeof watchOptions.poll === number 
        ? watchOptions.poll
        : undefined;
```

Reply:
oh yep we should respect user interval too

Reply:
fixed @alexander-akait